debezium-connector-jdbc/src/test/java/io/debezium/connector/jdbc/util/SinkRecordFactory.java

```diff
@@ -267,6 +268,45 @@ default SinkRecord createRecordWithSchemaValue(String topicName, byte key, List<
         return basicSchemaBuilder.build();
     }
 
+    default SinkRecord createInsertSchemaAndValue(String topicName, List<SchemaAndValueField> keyFields, List<SchemaAndValueField> valueFields, int offset) {
+
+        Schema keySchema = null;
+        if (!keyFields.isEmpty()) {
+            SchemaBuilder keySchemaBuilder = SchemaBuilder.struct();
+            for (SchemaAndValueField keyField : keyFields) {
+                keySchemaBuilder.field(keyField.fieldName(), keyField.schema());
+            }
+            keySchema = keySchemaBuilder.build();
+        }
+
+        final SchemaBuilder recordSchemaBuilder = SchemaBuilder.struct();
+        keyFields.forEach(keyField -> recordSchemaBuilder.field(keyField.fieldName(), keyField.schema()));
+        valueFields.forEach(valueField -> recordSchemaBuilder.field(valueField.fieldName(), valueField.schema()));
+        final Schema recordSchema = recordSchemaBuilder.build();
+
+        SinkRecordBuilder.SinkRecordTypeBuilder builder = SinkRecordBuilder.create()
+                .flat(isFlattened())
+                .name("prefix")
+                .topic(topicName)
+                .offset(offset)
+                .partition(0)
+                .keySchema(keySchema)
+                .recordSchema(recordSchema)
+                .sourceSchema(basicSourceSchema())
+                .source("ts_ms", (int) Instant.now().getEpochSecond());
+
+        for (SchemaAndValueField keyField : keyFields) {
+            builder.key(keyField.fieldName(), keyField.value());
```

Comment:
instead of doing this here and exposing this new API in SchemaAndValueField and adding another traversal over key fields, couldn't `SchemaBuilder#field()` do that already with an additional argument `boolean isKey` to distinguish between key and value fields?

Reply:
IMHO I am not a fan of methods with boolean arguments that behave differently based on it. So I'll maintain as is. 

Just for clearness I'll see a `keys` method that will get the list of key fields and do the loop. But it is just nitpick.

Reply:
I'd keep it as it is now and we might have a follow-up if needed.
@mfvitale You are right about the boolean argument yet in case of Kafka Connect it is pretty common that methods use `isKey` argument and change the behaviour based on that so at the end it would be consistent with current practice.

Reply:
I can relate those concerns, but I also find the performance gain is immense when you consider passing around millions and billions of messages over time. And you avoid iterating collections twice (or more).